File: DSA/problem_no_1.js
/* input =>nums=[int]
        target=int

output=> [int index array]

condition=> nums [index1]+nums[index2]=target */


var twosum=function(nums,target)
{
   for(let i=0;i<nums.length;i++)
   {
     let ele1=nums[i]
     let index1=i;
     let ele2=target-ele1
     if(nums.includes(ele2))
     {
        let index2=nums.indexOf(ele2)
        if(index1!=index2)
        {
            return [index1,index2]
        }
     }
   }
}

//let nums=[2,7,11,15]
let nums=[3,2,4]
//let target=9
let target=6
console.log(twosum(nums,target))

File: DSA/Problem_no_1.java

public class Problem_no_1{
    public static void main(String[] args) 
    {
        int [] nums={2,7,11,15};
        int target=9;
        int result[]=twosum(nums,target);
        if(result!=null)
        {
            for(int i=0;i<result.length;i++)
            {
                System.out.println(result[i]);
            }
        }
        else
        {
            System.out.println("no solution");
        }
       
    }
    public static int[] twosum(int[] nums, int target)
    {
        for(int i=0;i<nums.length;i++)
        {
            int ele1=nums[i];
            int index1=i;
            int ele2=target-ele1;
            for(int j=i+1;j<nums.length;j++)
            {
                if(nums[j]==ele2)
                {
                    int index2=j;
                    if(index1!=index2)
                    {
                        int [] result={index1,index2};
                        return result;
                    }
                }
                
            }
        }
        return null;
    }
}